specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-10-01-preview/mfe.json

```diff
@@ -10131,14 +10131,16 @@
             "name": "poolName",
             "description": "InferencePool name.",
             "required": true,
-            "type": "string"
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9\\-_]{0,254}$"
```

Comment:
what happens to the resources that exist today that do not follow this pattern ? 

same comment for all places.

Reply:
this is a new resource that just released in this API version, we actually have this check in server side, just missed in the swagger